Objects/genobject.c

```diff
@@ -342,8 +342,6 @@ _PyGen_yf(PyGenObject *gen)
 {
     if (gen->gi_frame_state == FRAME_SUSPENDED_YIELD_FROM) {
         _PyInterpreterFrame *frame = &gen->gi_iframe;
-        assert(is_resume(frame->instr_ptr));
```

Comment:
The old code was buggy... it couldn't handle an `ENTER_EXECUTOR` sitting on top of the `RESUME`.

Reply:
That makes `is_resume` buggy, not the assert, does it not? I think it would be better to fix `is_resume` to account for `ENTER_EXECUTOR`.

Similarly for the following assert.

Reply:
I guess I *can*, but I was just hesitant to because this `is_resume` function is used during generator finalization for an optimized fast path and I didn't want to slow it down too much with `ENTER_EXECUTOR` handling (which requires quite a bit of fishing around in the code object).

It's a fairly straightforward change to make if you think it's a good idea, though. Your call.

Reply:
I went ahead and fixed `is_resume`.